Makefile

```diff
@@ -37,6 +37,27 @@ ct-tls:
 edoc:
 	@rebar3 edoc skip_deps=true
 
+publish: edoc patch-doc
+	mix docs
+	mix hex.publish package
```

Comment:
I don't think generating the docs should be part of the publish target, because the generated `eredis.md` and should be checked in and committed before tagging and publishing to hex.

Suggestion: Add the patching to `make edoc` to run after `rebar3 edoc`. Let `make publish` also run edoc, but then check that no files are modified (using git diff or git describe with the proper flags) and abort the publishing if any file is changed. This makes sure the generated docs are committed in the repo before publishing.

Reply:
I prefer to run those commands one by one, I probably won't use this...

Reply:
I think we should patch as part of `make edoc` and let the patched files be checked in. Why?

There's a risk of any of these events:

1. Publishing is done using plain mix, without using make. (Erlang/Elixir hackers don't always have make installed, especially not on Windows, etc.) Also, it is good for to be in control of the publish steps, to be able to undo, redo, each command, publish/republish docs separately, etc.
2. Edoc is generated with `rebar3 edoc`, without using make.

If the patched markdown files are checked in, (1) is not a problem anymore. (2) can be easily spotted in code review. (If that's not good enough, we can try to add some hook to rebar to run a script after edoc.)

Reply:
Changed so that we check in the patched docs, and make sure the publish target checks for modified files.